provisioning_templates/PXEGrub2/kickstart_default_pxegrub2.erb

```diff
@@ -26,14 +26,26 @@ set default=0
 set timeout=<%= host_param('loader_timeout') || 10 %>
 
 menuentry '<%= template_name %>' {
-  <%= linuxcmd %> <%= @kernel %> ks=<%= foreman_url('provision') %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %> BOOTIF=01-$net_default_mac
+  <%= linuxcmd %> <%= @kernel %> ks=<%= foreman_url('provision') %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>
   <%= initrdcmd %> <%= @initrd %>
 }
 
-# Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1811561
-menuentry '<%= template_name %> HTTP Boot' {
-  <%= linuxcmd %> /httpboot/<%= @kernel %> ks=<%= foreman_url('provision') %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %> BOOTIF=01-$net_default_mac
-  <%= initrdcmd %> /httpboot/<%= @initrd %>
+<%
+if @host.provision_interface.subnet.httpboot
+  proxy_http_port = @host.provision_interface.subnet.httpboot.setting(:HTTPBoot, "http_port")
+  proxy_https_port = @host.provision_interface.subnet.httpboot.setting(:HTTPBoot, "https_port")
+  # workaround around "no DNS server configured" in grub: https://bugzilla.redhat.com/show_bug.cgi?id=1842509
+  proxy_host = dns_lookup(@host.provision_interface.subnet.httpboot.hostname)
```

Comment:
It looks like grub is not able to initialize DNS or parsing of hostname with port (`satellite:8000`) does not resolve correctly. This hack does allow the workflow, without it it would not be possible. I suggest to go with this and we can remove it later once the BZ is fixed. https://bugzilla.redhat.com/show_bug.cgi?id=1842509